vignettes/function-flow.Rmd

```diff
@@ -174,3 +173,112 @@ Finally, the dates outside of the data coverage period are dropped with
 `drop_diagnosis_dates_outside_coverage()` to end with a final diagnosis
 date. For details on this censoring based on periods with insufficient
 data coverage, see the `vignette("algorithm-logic")`.
+
+## Classifying the diabetes type
+
+The next step of the OSDC algorithm classifies individuals from the
+extracted diabetes population as having either T1D or T2D. The output is
+a `data.frame` that includes one row per individual in the diabetes
+population: one column with their PNR, two columns with inclusion dates
+(one "stable" date and one "raw" date - see the
+`vignette("algorithm_logic")` vignette for an elaboration on what that
+entails), and one column with the diabetes type.
+
+<!-- TODO: add a link to the specific section where this is described -->
+
+### Type 1 classification
+
+The details for the classification of type 1 diabetes is described in `vignette("design")`.
+Functions for the above filters are described in the following sections.
+
+![Flow of functions for classifying the diabetes type using the `osdc`
+package. The dark grey box denotes the user-facing function, while the
+remaining boxes are internal
+functions.](images/classify-diabetes-type-functions.png)
+
+#### Filter 1: Any T1D diagnoses and all GLD purchases are insulins
+
+This filter contains two functions:
+
+1.  `t1d_include_at_least_one_t1d_primary_diagnosis`, which relies on
+    the hospital diagnoses extracted from `lpr_diag` (LPR2) and
+    `diagnoser` (LPR3) in the previous steps.
+2.  `t1d_include_only_purchased_insulins` which relies on the GLD
+    purchases from Lægemiddelsdatabasen.
+
+#### Filter 2: Majority of T1D primary diagnoses, insulin within 180 days of diagnosis, and insulin constitutes 2/3 of all GLD doses
+
+The second filter includes three criteria: 1) Whether individuals have a
+majority of T1D primary diagnoses, 2) whether they purchased insulin
+within 180 days of diagnosis, and 3) whether insulin constitutes 2/3 of
+all their GLD doses.
+
+This results in three functions:
+
+1.  `t1d_include_majority_of_t1d_primary_diagnoses` (as compared to T2D
+    diagnoses) which again relies on primary hospital diagnoses from
+    DNPR.
+2.  `t1d_include_insulin_purchase_within_180_days_of_diagnosis` which
```

Comment:
and get_insulin_within_180_days() (for brevity at the cost of a little ambiguity)?